src/blockdigest.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "blockdigest.h"
+#include "streams.h"
+#include "version.h"
+#include "uint256.h"
+#include "primitives/transaction.h"
+#include "utiltime.h"
+#include "utilstrencodings.h"
+
+namespace DigestFilter {
+
+std::vector<uint8_t> base_filter::outpoint_data(const COutPoint& outpoint) const
+{
+    CDataStream stream(SER_NETWORK, PROTOCOL_VERSION);
+    stream << outpoint;
+    return std::vector<uint8_t>(stream.begin(), stream.end());
+}
+
+std::vector<const std::vector<uint8_t>> base_filter::tx_data_vec(const CTransaction& tx) const
+{
+    std::vector<const std::vector<uint8_t>> tx_data;
```

Comment:
Hm.. I don't know how many entries there are due to arbitrary pushes in SPKs and stuff, right? There could be several. Even with one, example output (with printfs) is
```
GetBFD 529
- adding 2ec87dce261d7a935202c3b0ba0b2e8db0fac3b81d631787aa5395e8e54a3727
- adding OP 2ec87dce261d7a935202c3b0ba0b2e8db0fac3b81d631787aa5395e8e54a3727:0
- adding data: 03ba8bbf669b83876fac1aa0fbdb24f19da0c5ad89dab673fc2f8ee3c3f7206c13
- adding OP 2ec87dce261d7a935202c3b0ba0b2e8db0fac3b81d631787aa5395e8e54a3727:1
- adding data: aa21a9ede2f61c3f71d1defd3fa999dfa36953755c690689799962b48bebd836974e8cf9
- adding in data 1102
- adding in data 01
```
which is 7 entries (2 last ones most likely should not exist but..), but tx has only 1 vin (coinbase) and 2 vouts.